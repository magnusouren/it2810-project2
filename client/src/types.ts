export interface User {
  name: string;
  id: string;
  loginState: boolean;
  ratings?: Rating[];
  watchlist?: Movie[];
}

/**
 * The Movie type is based on the API response from
 * https://developers.themoviedb.org/3/movies/get-popular-movies
 * The API response is a bit more complex, but we only need these fields.
 */
export interface Movie {
  adult: boolean;
  backdrop_path: string;
  genre_ids: number[];
  _id: number;
  original_language: string;
  original_title: string;
  overview: string;
  popularity: number;
  poster_path: string;
  release_date: string;
  title: string;
  video: boolean;
  vote_average: number;
  vote_count: number;
}

/**
 * The Category type is based on the API response from
 * https://developers.themoviedb.org/3/genres/get-movie-list
 */
export interface Category {
  id: number;
  name: string;
}

export interface Rating {
  movieId: number;
  rating: number;
}

export type AlphabeticalSort = 'a-z' | 'z-a' | '';

export type RatingSort = 'h-l' | 'l-h' | '';
