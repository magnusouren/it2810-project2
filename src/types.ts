export type User = {
  name: string;
  movieList?: Movie[];
};

export type Movie = {
  id: number;
  title: string;
  year: number;
  rating: number;
  poster: string;
};
