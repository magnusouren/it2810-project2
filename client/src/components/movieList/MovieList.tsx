import { FC } from 'react';

import { Movie } from '../../types';
import { MovieCard } from '../movieCard/MovieCard';
import styles from './MovieList.module.scss';

interface MovieListProps {
  movies: Movie[];
}

/**
 *
 * MovieList component.
 * Maps all movies to MovieCard components.
 *
 * @param movies
 * @returns {React.JSX.Element}
 *
 */

export const MovieList: FC<MovieListProps> = ({ movies }) => {
  return (
    <>
      <div className={styles.container}>
        {movies.map((movie) => (
          <MovieCard key={movie._id} movie={movie} />
        ))}
      </div>
    </>
  );
};
