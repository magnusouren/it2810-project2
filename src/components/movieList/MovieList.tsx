import { FC } from 'react';

import { Movie } from '../../types';
import { MovieCard } from '../movieCard/MovieCard';
import styles from './MovieList.module.scss';

interface MovieListProps {
  movies: Movie[];
}

export const MovieList: FC<MovieListProps> = ({ movies }) => {
  return (
    <div className={styles.container}>
      {movies.map((movie) => (
        <MovieCard key={movie.id} movie={movie} />
      ))}
    </div>
  );
};
