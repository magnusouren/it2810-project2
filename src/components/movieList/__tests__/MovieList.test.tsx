import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { Movie } from '../../../types';
import { MovieList } from '../MovieList';

describe('MovieList', () => {
  const movies: Movie[] = [
    {
      adult: false,
      backdrop_path: '/mockBackdropURL.jpg',
      genre_ids: [12, 28], // Adventure, Action
      id: 111111,
      original_language: 'en',
      original_title: 'Foo Bar',
      overview: 'Simple mock overview',
      popularity: 1000,
      poster_path: '/mockPosterURL.jpg',
      release_date: '1970-01-01',
      title: 'Foo Bar',
      video: false,
      vote_average: 5,
      vote_count: 555,
    },
    {
      adult: false,
      backdrop_path: '/mockBackdropURL.jpg',
      genre_ids: [12, 28], // Adventure, Action
      id: 222222,
      original_language: 'en',
      original_title: 'Foo Bar',
      overview: 'Simple mock overview',
      popularity: 1000,
      poster_path: '/mockPosterURL.jpg',
      release_date: '1970-01-01',
      title: 'Foo Bar',
      video: false,
      vote_average: 5,
      vote_count: 555,
    },
  ];

  it('should match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <MovieList movies={movies} />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render a MovieCard for each movie in the list', () => {
    render(
      <MemoryRouter>
        <MovieList movies={movies} />
      </MemoryRouter>,
    );

    const movieCards = screen.getAllByRole('link');
    expect(movieCards).toHaveLength(2);
  });
});
