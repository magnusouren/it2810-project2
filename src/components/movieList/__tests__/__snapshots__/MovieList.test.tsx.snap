// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieList > should match snapshot 1`] = `
<div>
  <div
    class="_container_e01689"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1gwrruu-MuiPaper-root-MuiCard-root"
    >
      <a
        class="_link_6d7af0"
        href="/movie/111111"
      >
        <img
          alt="Foo Bar"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          src="https://image.tmdb.org/t/p/w200/mockPosterURL.jpg"
        />
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          >
            Foo Bar
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Adventure, Action
          </p>
        </div>
      </a>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1gwrruu-MuiPaper-root-MuiCard-root"
    >
      <a
        class="_link_6d7af0"
        href="/movie/222222"
      >
        <img
          alt="Foo Bar"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          src="https://image.tmdb.org/t/p/w200/mockPosterURL.jpg"
        />
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
          >
            Foo Bar
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Adventure, Action
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;
